spring-xd-dirt/src/main/java/org/springframework/xd/dirt/modules/metadata/FtpSinkOptionsMetadata.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright 2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.xd.dirt.modules.metadata;
+
+import org.hibernate.validator.constraints.NotBlank;
+import org.springframework.xd.module.options.mixins.FtpConnectionMixin;
+import org.springframework.xd.module.options.spi.Mixin;
+import org.springframework.xd.module.options.spi.ModuleOption;
+
+/**
+ * Module options for FTP sink module.
+ *
+ * @author Franck Marchand
+ */
+@Mixin({ FtpConnectionMixin.class })
+public class FtpSinkOptionsMetadata {
+
+    private int clientMode = 0;
+
+    private String charset = "UTF-8";
+
+    private String remoteDir = "/";
+
+    private String temporaryRemoteDir = "/";
+
+    private boolean autoCreateDir = true;
+
+    private String remoteFileSeparator = "/";
+
+    private String tmpFileSuffix = ".tmp";
+
+    private String mode = "REPLACE";
+
+    private boolean useTemporaryFilename = true;
+
+    public boolean isUseTemporaryFilename() {
+        return useTemporaryFilename;
+    }
+
+    @ModuleOption("allows you to suppress using a temporary file name while writing the file")
+    public void setUseTemporaryFilename(boolean useTemporaryFilename) {
+        this.useTemporaryFilename = useTemporaryFilename;
+    }
+
+    @NotBlank
+    public String getMode() {
+        return mode;
+    }
+
+    @ModuleOption("what to do if the file already exists (enum FileExistsMode)")
```

Comment:
remove the reference to enum, this will be taken care of automatically. (see above)